### DISCUSSION TITLE
Clean way to include refraction when calculating separation_from between Apparent position vectors

### DISCUSSION THREAD

**Discussion by captcha1** [skyfielders/python-skyfield#735] [Ideas]
I'm looking for a clean way to include refraction when calculating separation_from between Apparent position vectors.

Perhaps an optional "temperature" argument passed to separation_from ?

Or a "refract" method in Apparent that returns a refracted position vector, then : 

lunardist = apparent0.refract("standard").separation_from(apparent1.refract("standard"))

FYI, this comes about when calculating "lunar distances" in celestial navigation.

Some gory details : I posted some skyfield code : http://fer3.com/arc/m2.aspx/Lunar-distance-calculations-Harris-mar-2022-g52292

... and Bill (who wrote the "Astron" celestial nav page) followed up with some test cases and fixes : https://pastebin.com/Rqb1YBsJ

Discuss ...

**Comment by brandon-rhodes** [comment]
First, thanks for following up with the solution you wound up going with!

Second, I have not in the past couple of months had time to sit down and really dwell on the question about whether there should be a more convenient way to refract the whole position, instead of getting back three separate coordinates (alt,az,distance) and then having to re-submit them to `from_altaz()` in the right order and assigned to the right keyword arguments. But it does seem a bit non-obvious and inconvenient. Skyfield buys a lot of simplicity by returning three individual coordinates from all of its "coordinate output" routines, but that does make maneuvers like the one you needed—turning those three _back_ into a Position—less happy.

I'll keep thinking about whether a `refract()` method on the position class might make sense someday. We'll see!
